packages/eslint-config-skuba/requireExtensions.js

```diff
@@ -0,0 +1,126 @@
+const { existsSync, lstatSync } = require('fs');
+const { dirname, resolve } = require('path');
+
+// Helper function to create rule listeners
+function createRuleListener(context, check) {
+  return {
+    DeclareExportDeclaration: (node) => processNode(node, context, check),
+    DeclareExportAllDeclaration: (node) => processNode(node, context, check),
+    ExportAllDeclaration: (node) => processNode(node, context, check),
+    ExportNamedDeclaration: (node) => processNode(node, context, check),
+    ImportDeclaration: (node) => processNode(node, context, check),
+  };
+}
+
+function extractPaths(inputPath) {
+  // Find all matches of '/src/' and their positions
+  const matches = [...inputPath.matchAll(/\/src\//g)];
+
+  if (matches.length === 0) {
+    return [process.cwd()]; // No '/src/' found in path, fallback to process.cwd()
+  }
+
+  const firstIndex = matches[0].index;
+  const lastIndex = matches[matches.length - 1].index;
+
+  return [
+    inputPath.substring(0, firstIndex + 1),
+    inputPath.substring(0, lastIndex + 1),
+  ];
+}
+
+function processNode(node, context, check) {
+  const source = node.source;
+  if (!source) {
+    return;
+  }
+  const value = source.value.replace(/\?.*$/, '');
+  if (!value || value.endsWith('.js')) {
+    return;
+  }
+
+  if (value.startsWith('.')) {
+    // Relative import, check if it ends with .js
+    return check(context, node, resolve(dirname(context.getFilename()), value));
+  }
+
+  if (value.startsWith('src')) {
```

Comment:
Handles our special `src` case. I think handling the others is considered under the too hard bucket.